holidays/countries/botswana.py

```diff
@@ -30,58 +29,61 @@ class Botswana(HolidayBase):
     country = "BW"
     special_holidays = {2019: ((JUL, 2, "Public Holiday"),)}
 
-    def _populate(self, year: int):
-        def _add_with_observed(
-            hol_date: date, hol_name1: str, hol_name2: str = None
-        ) -> None:
-            self[hol_date] = hol_name1
-            if hol_name2:
-                self[hol_date + td(days=+1)] = hol_name2
-
-            if self.observed and year >= 1995:
-                if self._is_saturday(hol_date) and hol_name2:
-                    self[hol_date + td(days=+2)] = f"{hol_name2} (Observed)"
-                elif self._is_sunday(hol_date):
-                    self[
-                        hol_date + td(days=+2 if hol_name2 else +1)
-                    ] = f"{hol_name1} (Observed)"
+    def __init__(self, *args, **kwargs) -> None:
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _add_observed(self, dt: date) -> None:
+        if self.observed and self._is_sunday(dt):
+            self._add_holiday(
+                "%s (Observed)" % self[dt],
+                dt + td(days=2 if dt + td(days=+1) in self else 1),
+            )
 
+    def _populate(self, year):
         if year <= 1965:
             return None
 
         super()._populate(year)
 
-        _add_with_observed(
-            date(year, JAN, 1), "New Year's Day", "New Year's Day Holiday"
+        jan_1 = self._add_new_years_day("New Year's Day")
+        self._add_observed(
+            self._add_new_years_day_two("New Year's Day Holiday")
         )
+        self._add_observed(jan_1)
 
         # Easter and easter related calculations
-        easter_date = easter(year)
-        self[easter_date + td(days=-2)] = "Good Friday"
-        self[easter_date + td(days=-1)] = "Holy Saturday"
-        self[easter_date + td(days=+1)] = "Easter Monday"
-        self[easter_date + td(days=+39)] = "Ascension Day"
+        self._add_good_friday("Good Friday")
+        self._add_holy_saturday("Holy Saturday")
+        self._add_easter_monday("Easter Monday")
+        self._add_ascension_thursday("Ascension Day")
 
-        _add_with_observed(date(year, MAY, 1), "Labour Day")
-        if self.observed and year >= 2016 and self._is_saturday(MAY, 1):
-            self[date(year, MAY, 1) + td(days=+2)] = "Labour Day Holiday"
+        may_1 = self._add_labor_day("Labour Day")
+        self._add_observed(may_1)
+        if self.observed and year >= 2016 and self._is_saturday(may_1):
+            self._add_holiday("Labour Day Holiday", may_1 + td(days=+2))
 
-        _add_with_observed(date(year, JUL, 1), "Sir Seretse Khama Day")
+        self._add_observed(self._add_holiday("Sir Seretse Khama Day", JUL, 1))
 
-        # 3rd Monday of July = "President's Day"
-        dt = _get_nth_weekday_of_month(3, MON, JUL, year)
-
-        self[dt] = "President's Day"
-        self[dt + td(days=+1)] = "President's Day Holiday"
+        jul_3_mon = self._add_holiday(
```

Comment:
Not sure about the best naming option for this. I guess one of `third_monday_of_july`, `july_3rd_monday` would be better here (being different from jul_3 approach).